src/compiler/scala/tools/nsc/backend/jvm/BTypes.scala

```diff
@@ -754,9 +754,7 @@ abstract class BTypes {
     /**
      * Pattern matching on a ClassBType extracts the `internalName` of the class.
      */
-    def unapply(c: ClassBType): Option[String] =
-      if (c == null) None
-      else Some(c.internalName)
+    def unapply(c: ClassBType): Some[String] = Some(c.internalName)
```

Comment:
I just checked in source code, did not find places that the null check is useful -- also usage of it is in the context of matching against `this`. So I assume this change is same.